src/inputfetcher.h

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2025-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INPUTFETCHER_H
+#define BITCOIN_INPUTFETCHER_H
+
+#include <attributes.h>
+#include <coins.h>
+#include <logging.h>
+#include <primitives/transaction_identifier.h>
+#include <tinyformat.h>
+#include <txdb.h>
+#include <util/hasher.h>
+#include <util/threadnames.h>
+
+#include <atomic>
+#include <barrier>
+#include <cstdint>
+#include <stdexcept>
+#include <thread>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+/**
+ * Helper for fetching inputs from the CoinsDB and CoinsTip and inserting them
+ * into the ephemeral cache used in ConnectBlock.
+ *
+ * It spawns a fixed set of worker threads that fetch Coins for each input
+ * in a block. The Coin is moved into the Input struct and then the status is
+ * atomically updated to READY. The main thread spin loops on the status field
+ * until it is READY and then inserts it into the temporary cache.
+ *
+ * Worker threads are synchronized with the main thread using a barrier, which
+ * is used at the beginning of fetching to start the workers and at the end to
+ * make sure all workers have exited the work loop.
+ */
+class InputFetcher
+{
+private:
+    //! The latest input being fetched. Workers atomically increment this when fetching.
+    alignas(64) std::atomic_size_t m_input_head{0};
```

Comment:
is the `alignas` a false sharing guard? Does it have a measurable effect?

Reply:
It was for false sharing. I don't think it has a measurable effect, but it might on some systems? I think it's harmless to keep, since there's only one InputFetcher.

Reply:
Removed it.